demo/IntrospectionModal.tsx

```diff
@@ -64,12 +67,27 @@ export class IntrospectionModal extends React.Component<IntrospectionModalProps>
     this.props.onClose();
   };
 
+  fetchIntrospection = (url, headers) => {
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        ...JSON.parse(headers),
+      },
+      body: JSON.stringify({ query: introspectionQuery }),
+    }).then((response) => response.json());
+  }
+
   handleSubmit = () => {
-    const { inputType, activePreset, jsonText, sdlText } = this.state.current;
+    const { inputType, activePreset, urlText, headers, jsonText, sdlText } = this.state.current;
     switch (inputType) {
       case Presets:
         this.props.onChange(PRESETS[activePreset]);
         break;
+      case URL:
+        this.fetchIntrospection(urlText, headers).then((data) => this.props.onChange(data));
```

Comment:
How UI look like during fetch?
and where is error handling?

Reply:
UI looks the same as for the rest "Introspection" and "SDL" tabs. Please note, there is no any error handling for them too. E.g.:

* If the user has entered invalid JSON or SDL into textarea, there is no any UI toast about JSON parsing or introspection from schema building errors.
* If the user has not entered JSON or SDL into textarea at all, the "Display" button is still active.

Errors are printed in the browser console.